hevc/mime.go

```diff
@@ -0,0 +1,60 @@
+package hevc
+
+import (
+	"fmt"
+)
+
+//Codecs - MIME subtype like hev1.1.6.L93.B0 where hev1 is sampleEntry string.
+//
+// Following ISO/IEC 14496-15 2017 Annex E
+func Codecs(sampleEntry string, sps *SPS) string {
+	profilePart := ""
+	ptl := sps.ProfileTierLevel
+	switch ptl.GeneralProfileSpace {
+	case 0:
+		// Nothing
+	case 1:
+		profilePart += "A"
+	case 2:
+		profilePart += "B"
+	case 3:
+		profilePart += "C"
+	}
+	profilePart += fmt.Sprintf("%d", ptl.GeneralProfileIDC)
+
+	flagsPart := fmt.Sprintf("%X", reverseFlags(ptl.GeneralProfileCompatibilityFlags))
+	var levelPart string
+	if ptl.GeneralTierFlag {
+		levelPart = "H"
+	} else {
+		levelPart = "L"
+	}
+	levelPart += fmt.Sprintf("%d", ptl.GeneralLevelIDC)
+	cif := ptl.GeneralConstraintIndicatorFlags
+	nrBytes := 6
+	for i := 0; i < 5; i++ { // Remove trailing zero bytes
+		if cif&0xff == 0 {
+			cif = cif >> 8
+			nrBytes--
+		} else {
+			break
+		}
+	}
+	// Code remaining nrBytes as hex separated by dots
+	constraintBytes := ""
+	for i := 0; i < nrBytes; i++ {
+		constraintBytes += fmt.Sprintf(".%X", (cif>>((nrBytes-1-i)*8))&0xff)
+	}
+
+	return fmt.Sprintf("%s.%s.%s.%s%s", sampleEntry, profilePart, flagsPart, levelPart, constraintBytes)
+}
+
+func reverseFlags(flags uint32) uint32 {
+	var rf uint32 = 0
+	for i := 0; i < 32; i++ {
+		if flags&(1<<(31-i)) != 0 {
+			rf |= 1 << i
+		}
+	}
+	return rf
+}
```

Comment:
Just realized that there is a Go package called `math/bits` which has the equivalent function that could reverse the bits: https://golang.org/pkg/math/bits/#Reverse32 

We could leverage the `math/bits` package to reduce the complexities.

Reply:
Yeah, that is a really good idea. In particular since it is part of the standard library so that it does not need any extra external package import.